syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.ev.fap.person";

import "date.proto";

package person;

message Person {
  string first_name = 2;
  string last_name = 3;
  //bytes picture = 4;
  bool is_verified = 5;
  //Height of the person in cms
  float height = 6;
  //List of phone numbers which are optional
  repeated string phone_numbers = 7;
  Gender gender = 8;
  utils.Date dob = 9;

  //Nested type
  message Address {
    string address_line_1 = 1;
    string address_line_2 = 2;
    string zip_code = 3;
    string city = 4;
    string country = 5;
  }

  repeated Address addresses = 10;
}

//Type defined in the same file
enum Gender {
  UNKNOWN = 0; //Default
  MALE = 1;
  FEMALE = 2;
}